pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -466,7 +466,7 @@ def test_agg_cython_category_not_implemented_fallback():
     result = df.groupby("col_num").col_cat.first()
     expected = pd.Series(
         [1, 2, 3], index=pd.Index([1, 2, 3], name="col_num"), name="col_cat"
-    )
+    ).astype("category")
```

Comment:
why is this changed?

Reply:
The `_cython_operation` method in this commit returns a categorical if passed a categorical. In this test the "col_cat" column is categorical so it seems intuitive to me that when aggregated over it should also be a category.

The fact that this test was passing before suggests that this is not also true in the higher order functions like `DataFrameGroupBy._cython_agg_blocks`.

Reply:
This change needs a dedicated whatsnew entry.

Agreed that preserving the dtype is probably the right choice, but this is a relatively large change.

Reply:
Sure, just added a new entry in the "Other API changes" section in commit 6498d6ba2